README.md

```diff
@@ -3080,6 +3079,172 @@ pinger = 10.5; now=true
+docker stop mqttwarn
+```
+
+#### Interactively
+
+If you want to experiment with your configuration, or to diagnose
+a problem, it can be simpler to restart the Python script,
+rather than to restart the Docker container.
+
+From the folder containing your `mqttwarn.ini` file:
+
+```
+docker run -it --rm \
+    -v $PWD:/opt/mqttwarn/conf \
+    --entrypoint bash \
+    jpmens/mqttwarn
+```
+ 
+To start the application: 
+```
+python mqttwarn.py
+```
+
+`Ctrl-C` will stop it. You can start and stop it as often as you like, here, probably editing the `.ini` file as you go.
+
+`Ctrl-D` or `exit` will stop the container. 
+
+
+
+#### Options
+
+##### Configuration Location
+
+You can of course run the Docker image from anywhere if you 
+specify a full path to the configuration file:
+```
+     -v /full/path/to/folder:/opt/mqttwarn/conf
+```
+
+##### Functions
+If you have one or more files of Python functions in the same folder
+as your `.ini` file, then prefix
+ the filenames in `.ini` file with a folder:
+```
+functions = 'functions/funcs.py'
+```
+Then add this argument to `docker run`:
+```
+    -v $PWD:/opt/mqttwarn/functions
+```
+
+##### Log file
+
+By default the log file will be created inside the container.
+If you would like instead log to a file on the host, add this to your
+`mqttwarn.ini` file:
+```
+logfile = 'log/mqttwarn.log'
+```
+Add this argument to `docker run`
+```
+    -v $PWD:/opt/mqttwarn/log
+```
+`mqttwarn.log` will be created in your current folder,
+and appended to
+each time the container is executed. You can delete the file
+between executions.
+
+
+##### If your MQTT Broker is Also Running in Docker on the Same Host
+
+If you give the MQTT broker container a name, then you can 
+refer to it by name rather than by
+IP address.  For instance, if it's named `mosquitto` 
+ put this in your `mqttwarn.ini` file:
+```
+hostname  = 'mosquitto'
+```
+Then add this argument to `docker run`:
+```
+    --link mosquitto
+```
+
+##### A full example
+
+If you have your `.ini` and Python files in your current directory,
+this will run `mqttwarn` and place the log file in the current directory:
+```
+docker run -d --rm --name mqttwarn \
+    -v $PWD:/opt/mqttwarn/conf \
+    -v $PWD:/opt/mqttwarn/functions \
+    -v $PWD:/opt/mqttwarn/log \
+    --link mosquitto \
+    jpmens/mqttwarn
+```
+
+
+
+### Build the image
+
+If you are making any changes to the `mqttwarn` application or to
+the `Dockerfile`, you can build a local image from the files on your drive
+(not from the files on Github).
+
+Execute the following from the root of the project :
+
+```
+docker build -t mqttwarn-local .
+```
+
+You can then edit any files and rebuild the image as many times as you need. 
+You don't need to commit any changes.
+
+The name `mqttwarn-local` is not meaningful, other than
+making it obvious when you run it that you are using your
+own personal image.  It will be renamed when it is published.
+
+### Deploy the Image (Administrators Only)
```

Comment:
This doesn't seem an entirely appropriate location. At some point we'll have a separate document for contributors rather than users?

Reply:
Yes